aspnet/fundamentals/routing.rst

```diff
@@ -1,10 +1,272 @@
+
+.. list-table:: Route Constraints
+	:header-rows: 1
+
+	* - constraint
+	  - Example
+	  - Example Match
+	  - Notes
+	* - ``int``
+	  - {id:int}
+	  - 123
+	  - Matches any integer
+	* - ``bool``
+	  - {active:bool}
+	  - true
+	  - Matches ``true`` or ``false``
+	* - ``datetime``
+	  - {dob:datetime}
+	  - 2016-01-01
+	  - Matches a valid ``DateTime`` value
+	* - ``decimal``
+	  - {price:decimal}
+	  - 49.99
+	  - Matches a valid ``decimal`` value
+	* - ``decimal``
+	  - {price:double}
+	  - 4.234
+	  - Matches a valid ``double`` value
+	* - ``float``
+	  - {price:float}
+	  - 3.14
+	  - Matches a valid ``float`` value
+	* - ``guid``
+	  - {id:guid}
+	  - 7342570B-44E7-471C-A267-947DD2A35BF9
+	  - Matches a valid ``Guid`` value
+	* - ``long``
+	  - {ticks:long}
+	  - 123456789
+	  - Matches a valid ``long`` value
+	* - ``minlength(value)``
+	  - {username:minlength(5)}
+	  - steve
+	  - String must be at least 5 characters long.
+	* - ``maxlength(value)``
+	  - {filename:maxlength(8)}
+	  - somefile
+	  - String must be no more than 8 characters long.
+	* - ``length(min,max)``
+	  - {filename:length(4,16)}
+	  - Somefile.txt
+	  - String must be at least 8 and no more than 16 characters long.
+	* - ``min(value)``
+	  - {age:min(18)}
+	  - 19
+	  - Value must be at least 18.
+	* - ``max(value)``
+	  - {age:max(120)}
+	  - 91
+	  - Value must be no more than 120.
+	* - ``range(min,max)``
+	  - {age:max(18,120)}
+	  - 91
+	  - Value must be at least 18 but no more than 120.
+	* - ``alpha``
+	  - {name:alpha}
+	  - Steve
+	  - String must consist of alphabetical characters.
+	* - ``regex(expression)``
+	  - {ssn:regex(\d{3}-\d{2}-\d{4})}
+	  - 123-45-6789
+	  - String must match the provided regular expression.
+	* - ``required``
+	  - {name:required}
+	  - Steve
+	  - Denotes a parameter is required; you need not specify this since it is the default.
+	* - ``any|regex|value``
+	  - {foo:a|b|c}
+	  - a
+	  - If the constraint isn't known, it is treated as a regex (if provided in the constraints dictionary - not within a template)
+
+.. tip:: To constrain a parameter to a known set of possible values, you can use a regex: ``{action:regex(list|get|create)}``. This would only match the ``action`` route value to ``list``, ``get``, or ``create``. If passed into the constraints dictionary, the string "list|get|create" would be equivalent.
+
+Constraints can be *chained*, so you can specify that a route value is of a certain type and also must fall within a certain range, like so: ``{age:int:range(1,120)}``. You can use the ``*`` character as a prefix to a route value name to bind to the rest of the URI. For example, ``blog/{*slug}`` would match any URI that started with ``/blog/`` and had any value following it (which would be assigned to the ``slug`` route value.
+
+Route templates must be unambiguous, or they will be ignored. For example, consider these two routes:
+
+1. ``{blog}/{year}/{month=1}``
+2. ``{blog}/{year}``
```

Comment:
Ok, I was having trouble coming up with a concrete example for the "route templates must be unambiguous" rule. Can you provide one instead of this one?